packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1930,24 +1931,38 @@ export default (superClass: Class<Parser>): Class<Parser> =>
 
       let accessibility: ?N.Accessibility;
       let readonly = false;
+      let override = false;
       if (allowModifiers !== undefined) {
-        accessibility = this.parseAccessModifier();
-        readonly = !!this.tsParseModifier(["readonly"]);
-        if (allowModifiers === false && (accessibility || readonly)) {
+        const modified = {};
+        this.tsParseModifiers(modified, [
+          "public",
+          "private",
+          "protected",
+          "override",
+          "readonly",
+        ]);
+        accessibility = modified.accessibility;
+        override = modified.override;
+        readonly = modified.readonly;
```

Comment:
I used `tsParseModifiers` on an empty object to reuse the logic to enforce the order. What do you think?